driver/docker-container/driver.go

```diff
@@ -380,9 +387,19 @@ func (d *Driver) Client(ctx context.Context) (*client.Client, error) {
 		}
 		return conn, nil
 	}))
+
 	if td, ok := exp.(client.TracerDelegate); ok {
 		opts = append(opts, client.WithTracerDelegate(td))
 	}
+
+	if len(co.Meta) > 0 {
+		opts = append(opts, client.WithGRPCDialOption(grpc.WithChainUnaryInterceptor(func(ctx context.Context, method string, req, reply interface{}, cc *grpc.ClientConn, invoker grpc.UnaryInvoker, opts ...grpc.CallOption) error {
+			// merge the existing context with new metadata.
+			ctx = metadata.NewOutgoingContext(ctx, co.Meta)
```

Comment:
I think it's safe to make a new set of metadata but let me know if you prefer to append instead for extra safety.

Reply:
Actually it's not safe, we need to append as we are joining metadata in buildkit like `buildkit-attachable-store-id`. Opened https://github.com/moby/buildkit/pull/4175.